chimney/src/main/scala/io/scalaland/chimney/dsl.scala

```diff
@@ -1,24 +1,30 @@
 package io.scalaland.chimney
 
+import shapeless.{HList, HNil, Witness, ::}
+
 object dsl {
 
   implicit class TransformerOps[From](val source: From) extends AnyVal {
 
-    def transformer[To]: TransformTo[From, To] = new TransformTo(source)
+    def transformer[To]: TransformTo[From, To, HNil] = new TransformTo(source, HNil)
   }
 
-  class TransformTo[From, To](val source: From) extends AnyVal {
+  class TransformTo[From, To, Modifiers <: HList](val source: From,
+                                                  val modifiers: Modifiers) {
 
-    def invoke(implicit transformer: Transformer.Aux[From, To]): To =
-      transformer.transform(source, Modifier.empty)
+    def withFieldValue[T](label: Witness.Lt[Symbol],
+                          value: T) = withFieldFunction(label, _ => value)
 
-    def invokeM[M <: Modifier](modifier: M)
-                              (implicit transformer: Transformer[From, To, M]): To =
-      transformer.transform(source, modifier)
-  }
+    def withFieldFunction[T](label: Witness.Lt[Symbol],
+                             f: From => T): TransformTo[From, To, Modifier.fieldFunction[label.T, From, T] :: Modifiers] =
+      new TransformTo(source, new Modifier.fieldFunction[label.T, From, T](f) :: modifiers)
 
-  type Modifier = io.scalaland.chimney.Modifier
-  val Modifier = io.scalaland.chimney.Modifier
+    def withRelabelled(label1: Witness.Lt[Symbol],
+                       label2: Witness.Lt[Symbol]): TransformTo[From, To, Modifier.relabel[label1.T, label2.T] :: Modifiers] =
+      new TransformTo(source, new Modifier.relabel[label1.T, label2.T] :: modifiers)
 
+    def invoke(implicit transformer: DerivedTransformer[From, To, Modifiers]): To =
```

Comment:
On a side note - if user do not want to provide anything we could combine `.into[To].convert` as `.convertInto[T]`